### PR TITLE
Added/Refactor: Clean MySql Queries in SugarFolders

### PR DESCRIPTION
**Pull request by code-ph0y**
Fix For My Account Loading Error #5506

## Description

This fix sanitises the data being entered into the queries allowing for special characters to be escaped  

## Motivation and Context

Because it is a security issue and will allow saving data correctly.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical 169reasons, the Appropriate Legal Notices must
```
Comment (pgorod) [inline tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php]: "technical 169reasons"?

File: tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+
+        $rootNode = new ExtNode('', '');
+
+        $folderOpenState = $user->getPreference('folderOpenState', 'Emails');
+        $folderOpenState = empty($folderOpenState) ? '' : $folderOpenState;
+
+        try {
+            $parentFolderOne->getUserFolders($rootNode, sugar_unserialize($folderOpenState), null, true);
+        } catch(SugarFolderEmptyException $e) {
+
```
Comment (gymad) [inline tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php]: :scream: Ok, this test passing when your function throws a SugarFolderEmptyException. but also passing if it doesn't?!

File: include/SugarFolders/SugarFolders.php [hunk 3]
```diff
@@ -42,21 +42,26 @@
...
 
 /**
  * Polymorphic buckets - place any item in a folder
  */
 class SugarFolder
 {
-
-    // public attributes
+    // Protected globals
+    protected $currentUser     = null;
```
Comment (gymad) [inline include/SugarFolders/SugarFolders.php]: please use @var [type] in phpdoc, above the variable.

File: tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php (hunk 1 above)
Comment (gymad) [reply tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php]: :open_mouth:

File: include/SugarFolders/SugarFolders.php [hunk 4]
```diff
@@ -510,71 +614,72 @@ public function getCountUnread($folderId)
 
     /**
      * Convenience method, pass a SugarBean and User to this to add anything to a given folder
+     *
+     * @param string $bean
+     * @param User   $user
+     * @return boolean
      */
     public function addBean($bean, $user = null)
```
Comment (gymad) [inline include/SugarFolders/SugarFolders.php]: You could remove unused $user parameter as it does not needed and passed from nowhere..

File: include/SugarFolders/SugarFolders.php (hunk 3 above)
Comment (code-ph0y) [reply include/SugarFolders/SugarFolders.php]: Fixed

File: include/SugarFolders/SugarFolders.php [hunk 5]
```diff
@@ -613,13 +625,13 @@ public function getCountUnread($folderId)
...
-     * Convenience method, pass a SugarBean and User to this to add anything to a given folder
+     * Convenience method, pass a SugarBean
      *
      * @param string $bean
-     * @param User   $user
+     *
      * @return boolean
      */
-    public function addBean($bean, $user = null)
+    public function addBean($bean)
```
Comment (gymad) [inline include/SugarFolders/SugarFolders.php]: I don't think the $bean should be a string. Use SugarBean type hinting and corrigate the phpdoc above please.

**Pull request merged**